### DISCUSSION TITLE
Newbie Question: Using GitHub to build Badger 2040W image with BLE

### DISCUSSION THREAD

**Discussion by DuaneKaufman** [micropython/micropython#11822] [RP2040 / Pico] [answered]
Dear All,
I am ignorant about how to do the following: I have a Badger 2040W (https://learn.pimoroni.com/article/getting-started-with-badger-2040) and I would like to use BLE, _and_ the display.
Pimoroni builds custom firmware by pulling the rp2 Micropython code, and incorporating their own graphics display library (among other things) - (https://github.com/pimoroni/badger2040/blob/main/.github/workflows/micropython.yml)
I've tested my Badger Pico 2040W with a Micropython nightly build, and BLE seems to be working fine.
I would like to use a nightly Micropython build (to get BLE) and couple it with the Badger Pico 2040W libraries to get all of the functionality running (BLE with display). From the Pimoroni file (above) it sounds like I should be able to pull a nightly build of Micropython, and used GitHub Actions to create a .UF2 file, but I'm really in deep water, and have never used Github to do this before.
If anyone has tips/comments, they would be welcome.

Sincerely,
Duane

**Comment by andypiper** [comment, marked as answer]
It looks like Pimoroni are in the process of adding BLE into their own builds... https://github.com/pimoroni/pimoroni-pico/releases/tag/v1.20.3 (I assume the build for the Badger will include it)

**Reply by DuaneKaufman** [reply]
Hi,
I see that is pretty recent, but good news! The Pimoroni announcement indicates they will be using Micropython release  856e08b1931b88271816a2f60648f6ff332235b2. How do I find that in the Micropython repo?

Thanks again.
Duane
